### ISSUE TITLE
Incorrect no-member for super().__getattr__

### ISSUE THREAD

**Issue by The-Compiler** [PyCQA/pylint#979]
With python 3, this code is valid:

```python
class Foo:

    def __getattr__(self, attr):
        return super().__getattr__(attr)
```

With both 1.5.6 and the current master of pylint/astroid I get:

```
E:  4,15: Super of 'Foo' has no '__getattr__' member (no-member)
```

**Comment by PCManticore** [comment]
Does the parent of Foo actually has a ``__getattr__`` method or is this class is intended to be used as a mixin? If it is the latter case, when inferring super(), we don't know if it is supposed to be used as a mixin, so we look only in the parents of the class where the call lies, through the method resolution order and then we check where is that member, if it actually exists somewhere. You can probably change this behavior by renaming your class to FooMixin or something of sorts.

**Comment by The-Compiler** [comment]
I assumed it inherits `__getattr__` from `object`?

**Comment by PCManticore** [comment]
object does not have a ``__getattr__``, it is only defined by user classes and processed by Python in certain conditions. It has a ``__getattribute__`` though.

**Comment by PCManticore** [comment]
For instance, in this example, we have the following crash:

```
$ cat d.py
    class C:
       def __getattr__(self, attr):
             1/0
             return super().__getattr__(attr)

    C().__reduce_ex__
    C().x

  $ py -3 d.py
    Traceback (most recent call last):
      File "d.py", line 7, in <module>
        C().x
      File "d.py", line 3, in __getattr__
        1/0
    ZeroDivisionError: division by zero
```
This happens because the ``__reduce_ex__`` is already found by ``__getattribute__``, before triggering ``__getattr__``, so this fails at runtime in case where the attribute is not already defined by the type of the object.

**Comment by PCManticore** [comment]
I presume this is the actual chain of operations, I am not that familiar with the attribute lookup, so feel free to correct me.

**Comment by The-Compiler** [comment]
I think you're right - attribute lookup really confuses me as well.

Then again I don't get why my code here works without any issues...

```python
class TabData:

    """A simple namespace with a fixed set of attributes.

    Attributes:
        keep_icon: Whether the (e.g. cloned) icon should not be cleared on page
                   load.
        inspector: The QWebInspector used for this webview.
    """

    def __init__(self):
        self._data = {
            'keep_icon': False,
            'viewing_source': False,
            'inspector': None,
        }

    def __getattr__(self, attr):
        if attr.startswith('_'):
            # WORKAROUND for https://github.com/PyCQA/pylint/issues/979
            return super().__getattr__(attr)  # pylint: disable=no-member
        try:
            return self._data[attr]
        except KeyError:
            raise AttributeError(attr)

    def __setattr__(self, attr, value):
        if attr.startswith('_'):
            return super().__setattr__(attr, value)
        if attr not in self._data:
            msg = "Can't set unknown attribute {!r}".format(attr)
            raise AttributeError(msg)
        self._data[attr] = value
```

```python
class TestTabData:

    def test_known_attr(self):
        data = tab.TabData()
        assert not data.keep_icon
        data.keep_icon = True
        assert data.keep_icon

    def test_unknown_attr(self):
        data = tab.TabData()
        with pytest.raises(AttributeError):
            data.bar = 42
        with pytest.raises(AttributeError):
            data.bar  # pylint: disable=pointless-statement
```

**Comment by PCManticore** [comment]
But it doesn't. ;-) For instance, try with ``data.__test__`` and you will see why. Every other dunder attribute, such as ``__class__`` for instance, does not even reach that point, because it will be retrieved through ``__getattribute__`` or something of sorts.

**Comment by The-Compiler** [comment]
Oh, indeed! Seems like the `super()`-dance is only required for `__setattr__`, and as you said, for getting attributes it's `__getattribute__` which will be invoked. Thanks for the explanations!
